tests/wamr-test-suites/test_wamr.sh

```diff
@@ -394,117 +396,83 @@ function spec_test()
     touch ${REPORT_DIR}/spec_test_report.txt
 
     cd ${WORK_DIR}
-    if [ ! -d "spec" ];then
-        echo "spec not exist, clone it from github"
-        git clone -b master --single-branch https://github.com/WebAssembly/spec
-    fi
-
-    pushd spec
-
-    # restore and clean everything
-    git reset --hard HEAD
 
     # update basic test cases
-    echo "update spec test cases"
-    git fetch origin main
-    # restore from XX_ignore_cases.patch
-    # resotre branch
-    git checkout -B main
-    # [spec] Update note on module initialization trapping (#1493)
-    git reset --hard 044d0d2e77bdcbe891f7e0b9dd2ac01d56435f0b
-    git apply ../../spec-test-script/ignore_cases.patch
-    if [[ ${ENABLE_SIMD} == 1 ]]; then
-        git apply ../../spec-test-script/simd_ignore_cases.patch
-    fi
-    if [[ ${ENABLE_MULTI_MODULE} == 1 && $1 == 'aot'  ]]; then
-        git apply ../../spec-test-script/multi_module_aot_ignore_cases.patch
-    fi
+    echo "downloading spec test cases..."
 
-    # udpate thread cases
+    rm -rf spec
     if [ ${ENABLE_MULTI_THREAD} == 1 ]; then
-        echo "checkout spec for threads proposal"
-        if [[ -z $(git remote -v | grep "\<threads\>") ]]; then
-            git remote add threads https://github.com/WebAssembly/threads
-        fi
+        echo "checkout spec from threads proposal"
 
-        # fetch spec for threads proposal
-        git fetch threads
-        # Fix error in Web embedding desc for atomic.notify (#185)
-        git reset --hard 85b562cd6805947876ec5e8b975ab0127c55a0a2
-        git checkout threads/main
-
-        git apply ../../spec-test-script/thread_proposal_ignore_cases.patch
-        git apply ../../spec-test-script/thread_proposal_fix_atomic_case.patch
-    fi
+        # check spec test cases for threads
+        git clone -b main --single-branch https://github.com/WebAssembly/threads.git spec
+        pushd spec
 
-    if [ ${ENABLE_EH} == 1 ]; then
+        # [interpreter] implement atomic.wait and atomic.notify (#194)
+        git reset --hard 09f2831349bf409187abb6f7868482a8079f2264
+        git apply ../../spec-test-script/thread_proposal_ignore_cases.patch || exit 1
+        git apply ../../spec-test-script/thread_proposal_fix_atomic_case.patch || exit 1
+    elif [ ${ENABLE_EH} == 1 ]; then
         echo "checkout exception-handling test cases"
-        popd
-        if [ ! -d "exception-handling" ];then
-            echo "exception-handling not exist, clone it from github"
-            git clone -b master --single-branch https://github.com/WebAssembly/exception-handling 
-        fi
-        pushd exception-handling
+
+        # TBC: master -> main?
+        git clone -b master --single-branch https://github.com/WebAssembly/exception-handling spec
+        pushd spec
 
         # restore and clean everything
         git reset --hard 51c721661b671bb7dc4b3a3acb9e079b49778d36
-        
+
         if [[ ${ENABLE_MULTI_MODULE} == 0 ]]; then
```

Comment:
Can remove this line since currently we don't test spec cases of exception handling when multi-module is enabled